File: ios/Classes/handler/CancelOperationCommand.swift
import Flutter
import Foundation

extension Notification.Name {
    static let operationCanceled = Notification.Name("operationCanceled")
}

/**
 * Command for canceling currently running operations
 */
class CancelOperationCommand: Command {
    func execute(call: FlutterMethodCall, result: @escaping FlutterResult) {
        OperationManager.shared.queue.async(flags: .barrier) {
            for (_, workItem) in OperationManager.shared.operations {
                workItem.cancel()
            }
            OperationManager.shared.operations.removeAll()
            NotificationCenter.default.post(name: .operationCanceled, object: nil)
            DispatchQueue.main.async {
                result(nil)
            }
        }
    }
}

File: android/src/main/kotlin/com/example/easy_video_editor/handler/CancelOperationCommand.kt
package com.example.easy_video_editor.handler

import com.example.easy_video_editor.command.Command
import com.example.easy_video_editor.utils.OperationManager
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

/**
 * Command for canceling currently running operations
 */
class CancelOperationCommand : Command {
    override fun execute(call: MethodCall, result: MethodChannel.Result) {
        try {
            val wasCanceled = OperationManager.cancelAllOperations()
            result.success(wasCanceled)
        } catch (e: Exception) {
            result.error(
                "CANCEL_ERROR",
                "Error canceling operations: ${e.message}",
                null
            )
        }
    }
}
